src/CollectionInterface.php

```diff
@@ -120,7 +120,7 @@ public function last();
      * @param string $order The sort order for the resulting collection (one of
      *     this interface's `SORT_*` constants).
      *
-     * @return CollectionInterface<T>
```

Comment:
Stating the return type of `CollectionInterface` instead of `static` means that any methods on the top level will be lost. For example:

```php
class FooCollection extends AbstractCollection
{
    public function getType(): string
    {
        return Foo::class;
    }

    public function allActive(): bool
    {
        return count($this) === count($this->filter(...));
    }
}
```

If I were to do `$x = $foo->map(...);` then my IDE would not see `$x->allActive()` as a possible method and show a warning.